src/sql/azdata.d.ts

```diff
@@ -2515,7 +2515,7 @@ declare module 'azdata' {
 	 * AccountProvider.refresh or AccountProvider.prompt are rejected with this error, the error
 	 * will not be reported to the user.
 	 */
-	export interface PromptFailedResult {
+	export interface ProviderError {
```

Comment:
We really shouldn't ever be updating stuff in azdata.d.ts unless it's really necessary - doing this will cause anyone that might be using this to have compilation failures when updating the typings.

I'd also say that the `canceled` property here makes this something more specific than just generic "ProviderError".

My suggestion here would be to undo the changes in this file, and then in azdata.proposed add this 

`export interface PromptFailedResult extends ProviderError { }` 

in addition to the addition of the new ProviderError interface you added. This way we can still use that error for generic provider errors (even if they know nothing about a "canceled" state) while keeping the existing typings intact.

Reply:
Do you mean to say `export interface ProviderError extends PromptFailedResult`? since I would be changing the one in azdata.d.ts back to PromptFailedResult

Reply:
oh I see I misunderstood, that makes sense.